### PR TITLE
nixos/modules: Nicer nice level handling

### PR DESCRIPTION
**Pull request by Pamplemousse**
Signed-off-by: Pamplemousse <xav.maso@gmail.com>

###### Motivation for this change

Bring some consistency, and make better use of type safety.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fricklerhandwerk** [comment]
LGTM

**Comment by i077** [comment]
Maybe there should be a type for niceness values (like we have one for ports)? That way it's more discoverable.

**Comment by Pamplemousse** [comment]
> That way it's more discoverable for other people writing modules.

Out of curiosity: More discoverable from where? And how (which tools, commands, or else)?

**Comment by i077** [comment]
> Out of curiosity: More discoverable from where? And how (which tools, commands, or else)?

Actually I'm not really aware of any tools that will enumerate the types in the nixpkgs library, I just thought module authors might see a type called `niceness` used in other modules and think "oh, I can use that for my option instead of `types.int`".

File: lib/types.nix [hunk 1]
```diff
@@ -451,6 +451,8 @@ rec {
       nestedTypes.elemType = elemType;
     };
 
+    niceness = types.ints.between (-20) 19;
```
Comment (i077) [inline lib/types.nix:454]: Sorry for the nit, but I think this'd be better placed near the `int` type (right below where `ports` is defined).

**Comment by Infinisil** [comment]
Please add a `mkRenamedOptionModule` for the renamed options.

File: nixos/modules/services/backup/btrbk.nix [hunk 2]
```diff
@@ -51,6 +51,10 @@ let
     '';
 in
 {
+  imports = [
+    (mkRenamedOptionModule [ "services" "btrbk" "niceness" ] [ "services" "btrbk" "daemonNiceLevel" ])
```
Comment (i077) [inline nixos/modules/services/backup/btrbk.nix:55]: This (and other uses of `mkRenamedOptionModule`) isn't defined in the file, you need to inherit it from the `lib` set.

Comment (i077) [inline nixos/modules/services/backup/btrbk.nix:55]: You need to grab this from the `lib` set before using it. I think this causes both tests to fail.

**Review by i077 — CHANGES_REQUESTED** [review]

**Review by i077 — APPROVED** [review]
LGTM

**Comment by Pamplemousse** [comment]
@Infinisil any other changes you would like to see?

**Pull request closed without merging**